ipn/ipnlocal/local.go

```diff
@@ -663,6 +664,12 @@ func (b *LocalBackend) linkChange(delta *netmon.ChangeDelta) {
 	b.updateFilterLocked(b.netMap, b.pm.CurrentPrefs())
 	updateExitNodeUsageWarning(b.pm.CurrentPrefs(), delta.New, b.health)
 
+	// Upon any link change, wait three seconds for things to settle, then asynchronously run
+	// captive portal detection.
+	time.AfterFunc(3*time.Second, func() {
+		b.performCaptiveDetection()
+	})
```

Comment:
This should probably be one time.Timer that's stored and reset (extended) on every change, so these don't pile up in a flood after a bunch of noisy link changes?